src/inmanta/deploy/scheduler.py

```diff
@@ -593,6 +695,74 @@ async def is_agent_running(self, name: str) -> bool:
+        """
+        Schedule individual periodic repair for a given resource iff the repair timer is specified as
+        an int
+        """
+
+        async def _repair() -> None:
+            async with self._scheduler_lock:
+                if self._state.resource_state[resource].blocked is BlockedStatus.YES:  # Can't deploy
+                    return
+                to_deploy: set[ResourceIdStr] = {resource}
+
+                self._work.deploy_with_context(
+                    to_deploy,
+                    reason=f"Individual repair triggered for resource {resource} because last "
+                    f"repair happened more than {self._repair_timer}s ago.",
+                    priority=TaskPriority.INTERVAL_REPAIR,
+                    deploying=self._deploying_latest,
+                )
+
+        assert isinstance(self._repair_timer, int)
+        interval_schedule: IntervalSchedule = IntervalSchedule(
+            interval=float(self._repair_timer), initial_delay=self._repair_timer
+        )
+        task = self._sched.add_action(_repair, interval_schedule)
+
+        # Omitting this yields mypy error:
+        # Returning Any from function declared to return "ScheduledTask | None"  [no-any-return]
+        assert task is None or isinstance(task, ScheduledTask)
+        return task
+
+    def create_recurring_deploy_for_resource(self, resource: ResourceIdStr) -> ScheduledTask | None:
+        """
+        Schedule individual periodic deploy for a given resource iff the deploy timer is specified as
+        an int
+        """
+
+        async def _deploy() -> None:
+            async with self._scheduler_lock:
+                if any(
+                    [
+                        self._state.resource_state[resource].status is ResourceStatus.UP_TO_DATE,  # No need to deploy
+                        self._state.resource_state[resource].blocked is BlockedStatus.YES,  # Can't deploy
+                    ]
+                ):
+                    return
+
+                to_deploy: set[ResourceIdStr] = {resource}
+
+                self._work.deploy_with_context(
+                    to_deploy,
+                    reason=f"Individual deploy triggered for resource {resource} because last "
+                    f"deploy happened more than {self._deploy_timer}s ago.",
+                    priority=TaskPriority.INTERVAL_DEPLOY,
+                    deploying=self._deploying_latest,
+                )
+
+        assert isinstance(self._deploy_timer, int)
+        interval_schedule: IntervalSchedule = IntervalSchedule(
+            interval=float(self._deploy_timer), initial_delay=self._deploy_timer
+        )
+        task = self._sched.add_action(_deploy, interval_schedule)
+
+        # Omitting this yields mypy error:
+        # Returning Any from function declared to return "ScheduledTask | None"  [no-any-return]
+        assert task is None or isinstance(task, ScheduledTask)
+        return task
```

Comment:
I wonder if we really want to run these timer for resources that are not in the dirty set. 

if we have, let's say 50k resources and a 10s deploy interval and an empty dirty set, we still process 5k timers per second for no reason.  (these numbers are high, but not crazy either)